src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -937,6 +938,53 @@ class ObjectIdGenerator : public Generator<bsoncxx::types::b_oid> {
     const UniqueGenerator<std::string> _hexGen;
 };
 
+/**
+ * `{^UUID: {hex: "3b241101-e2bb-4255-8caf-4136c566a962"} }`
+ *  Generator for non-legacy UUIDs (0x04). hex can itself be a generator so the string can be
+ *  generated in many fashions ^Join, ^FormatString, ... etc.
+ *
+ *  The input format accepted for the hex field is 8-4-4-4-12 format. The code is agnostic as to the
+ *  UUID version.
+ *
+ *  see https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)
+ */
+class UuidGenerator : public Generator<bsoncxx::types::b_binary> {
+public:
+    UuidGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _rng{generatorArgs.rng},
+          _node{node},
+          _hexGen{stringGenerator(node["hex"], generatorArgs)},
+          _subTypeGen{intGenerator(node["subType"],
```

Comment:
Could you move the initialization of _subTypeGen into the constructor body? Then you could test for existence of `node["subType"]` and pass in the default value if the node doesn't exist. Then you don't need to define the new version of intGenerator.  It feels like a lot of machinery to add to allow a default value, when only this generator is using a default value.

Reply:
Presumably you could stay with the inline definition by using a ternary expression here, but those always feel like magic to me.